mypyc/lib-rt/mypyc_util.h

```diff
@@ -31,6 +31,8 @@
 // Here just for consistency
 #define CPy_XDECREF(p) Py_XDECREF(p)
 
+#ifndef Py_GIL_DISABLED
+
 // The *_NO_IMM operations below perform refcount manipulation for
```

Comment:
Can you adjust this comment to mention something like "this optimization cannot be performed with free threaded mode <for x reasons> so we fallback to just calling the normal decref.